interfaces/matlab_experimental/Reactors/Reactor.m

```diff
@@ -0,0 +1,328 @@
+            % :param flag:
+            %    String, either "on" or "off" to enable or disable chemical
+            %    reactions, respectively.
+            %
+
+            if strcmp(flag, 'on')
+                iflag = true;
+            elseif strcmp(flag, 'off')
+                iflag = false;
+            else error('Input must be "on" or "off"');
+            end
+
+            callct('reactor_setChemistry', r.id, iflag);
+        end
+
+        function setEnergyFlag(r, flag)
+            % Enable or disable solving the energy equation.
+            %
+            % r.setEnergyFlag(flag)
+            %
+            % If the energy equation is disabled, then the reactor
+            % temperature is constant. The parameter should be the string
+            % "on" to enable the energy equation, or "off" to disable it.
+            %
+            % By default, Reactor objects are created with the energy
+            % equation enabled, so usually this method
+            %
+            % :param r:
+            %    Instance of class 'Reactor'.
+            % :param flag:
+            %    String, either "on" or "off" to enable or disable chemical
+            %    reactions, respectively.
+            %
+
+            iflag = -1;
+
+            if strcmp(flag, 'on')
+                iflag = 1;
+            elseif strcmp(flag, 'off')
+                iflag = 0;
+            end
+
+            if iflag >= 0
+                callct('reactor_setEnergy', r.id, iflag);
+            else error('Input must be "on" or "off".');
+            end
+
+        end
+
+        function setThermoMgr(r, t)
+            % Set the thermodynamics manager.
+            %
+            % r.setThermoMgr(t)
+            %
+            % This method is used internally during Reactor initialization,
+            % but is usually not called by users.
+            %
+            % :param r:
+            %    Instance of class 'Reactor'.
+            % :param t:
+            %    Instance of class 'ThermoPhase' or another object
+            %    containing an instance of that class.
+
+            if ~isa(t, 'ThermoPhase')
+                error('Wrong object type');
+            end
+
+            callct('reactor_setThermoMgr', r.id, t.tpID);
+        end
+
+        function setKineticsMgr(r, k)
+            % Set the kinetics manager.
+            %
+            % r.setKineticsMgr(k)
+            %
+            % This method is used internally during Reactor initialization,
+            % but is usually not called by users.
+            %
+            % :param r:
+            %    Instance of class 'Reactor'.
+            % :param t:
+            %    Instance of class 'Kinetics' or another object
+            %    containing an instance of that class.
+
+            if ~isa(k, 'Kinetics')
+                error('Wrong object type');
+            end
+
+            callct('reactor_setKineticsMgr', r.id, k.kinID);
+        end
```

Comment:
> Using a class in its intended way should not produce warnings. I would suggest just removing the ability to specify optional properties as part of the constructor, and rely on setting the ones that have non-default values in subsequent statements.



> So if I understand correctly, the preferred way would be for the class constructor to only create the object and set default values. Then the user would set those values outside of the class constructor?

In an earlier conversation with Ray (from July), he mentioned the preferred way for constructors is to just create the object with its ID and type and setting default values (in the case of `Reactor`, the default is an empty reactor with no content). The user then set properties/contents later. This is how the legacy Matlab toolbox handles constructors and a lot of the examples are based on this. Similar methods are also present in the Python interface (`cantera.ReactorSurface.install`). So for now I have kept these methods as is. 

That being said, in this case I think I should change setThermoMgr and setKineticsMgr as setter methods for `reactor.thermo` and `reactor.kinetics`.

Reply:
My comment was about the handling for all of the _optional_ arguments to the `Reactor` constructor. Since a `Reactor` cannot be used without a `ThermoPhase` and `Kinetics` object, I think it is perfectly reasonable to have a `Solution` object as a _required_ constructor argument.

Reply:
:+1: … I think there is value in avoiding necessary assignments after the creation of an object. Also, the way the old MATLAB interface worked should not be taken as a template. Rather, please use the Python API as a guide - while there have to be some differences, making the look and feel similar when possible will be beneficial to the community.